logging/exp/testing.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exp
+
+import (
+	"io"
+	"testing"
+
+	"golang.org/x/exp/slog"
+)
+
+// TestLogger creates a new logger for use in tests. It will only log messages
+// when tests fail and the tests were run with verbose (-v).
+func TestLogger(tb testing.TB) *slog.Logger {
```

Comment:
@jba this might be worth thinking about in the stdlib. It's super small so it's easy to DIY, but it could be a nice addition.

Reply:
Nice. There's been some call for that. Will consider as an add-on proposal.

Reply:
See discussion at https://github.com/golang/go/issues/56345#issuecomment-1295757032. I think your implementation will have the wrong source lines (from the tb.Log call) as well as the right ones.